torch/cuda/nccl.py

```diff
@@ -12,6 +12,8 @@
 def is_available(tensors):
     devices = set()
     for tensor in tensors:
+        if tensor.is_sparse:
+            return False
```

Comment:
It's not true that having a mix of sparse dense will prevent us from using NCCL, right? They will be different groups and dense will be still reduced with NCCL?

Reply:
My understanding is that `nccl.is_available` should be true only if NCCL can be applied to all tensors in `tensors`. It is responsibility of the user of this function to separate sparse and dense tensors. 

In fact, it is currently only used in `broadcast` and `reduce_add`. In both cases, reduced tensors are already either all sparse or all dense.